src/components/shared/TimedPopup.js

```diff
@@ -1,115 +1,78 @@
-
-  confirmTimeoutHandle = null;
-
-  shouldComponentUpdate(nextProps, nextState) {
-    const { message, downward, openOnInit, timeout, language, refElement, updateTrigger } = this.props;
-    const { opened }                                                                      = this.state;
+  const [opened, setOpened] = useState(false);
 
-    const toUpdate = openOnInit !== nextProps.openOnInit
-      || updateTrigger !== nextProps.updateTrigger
-      || message !== nextProps.message
-      || downward !== nextProps.downward
-      || language !== nextProps.language
-      || timeout !== nextProps.timeout
-      || refElement !== nextProps.refElement
-      || opened !== nextState.opened;
+  useEffect(() => {
+    let confirmTimeoutHandle = null;
 
-    return toUpdate;
-  }
-
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    if (this.props.updateTrigger !== nextProps.updateTrigger || this.props.openOnInit !== nextProps.openOnInit) {
-      if (nextProps.openOnInit) {
-        this.open();
-      } else {
-        this.setState({ opened: false });
+    const clearConfirmationTimeout = () => {
+      if (confirmTimeoutHandle) {
+        clearTimeout(confirmTimeoutHandle);
+        confirmTimeoutHandle = null;
       }
-    }
-  }
-
-  componentWillUnmount() {
-    this.clearConfirmationTimeout();
-  }
+    };
 
-  clearConfirmationTimeout = () => {
-    if (this.confirmTimeoutHandle) {
-      clearTimeout(this.confirmTimeoutHandle);
-      this.confirmTimeoutHandle = null;
+    if (openOnInit) {
+      clearConfirmationTimeout();
+      setOpened(true);
+      confirmTimeoutHandle = setTimeout(() => setOpened(false), timeout);
+    } else {
+      setOpened(false);
     }
-  };
 
-  onTrigger = () => {
-    const { onTrigger } = this.props;
-    if (onTrigger) {
-      onTrigger();
+    return () => {
+      clearConfirmationTimeout();
     }
-  };
+  }, [openOnInit, timeout, updateTrigger])
 
-  open = () => {
-    this.clearConfirmationTimeout();
-    this.setState({ opened: true }, () => {
-      this.confirmTimeoutHandle = setTimeout(
-        () => this.setState({ opened: false }),
-        this.props.timeout);
-    });
+  const style = {
+    position: 'absolute',
+    right: 0,
+    [downward ? 'top' : 'bottom']: 0,
   };
 
-  render() {
-    const { message, downward, language, refElement } = this.props;
-    const { opened }                                  = this.state;
-    const style                                       = {
-      position: 'absolute',
-      right: 0,
-      [downward ? 'top' : 'bottom']: 0,
-    };
-    if (downward) {
-      style.top = 0;
-    } else {
-      style.bottom = 0;
-    }
-
-    const rtlLang = isLanguageRtl(language);
-
-    return (
-      <Popup
-        open={opened}
-        content={message}
-        position={`${downward ? 'bottom' : 'top'} right`}
-        className={rtlLang ? 'change-popup-direction' : ''}
-        context={refElement}
-      />
-    );
+  if (downward) {
```

Comment:
Isn't in already initialized in line 46?

Reply:
no... it is a property... and asked for its value...